src/software/ai/navigator/obstacle/robot_navigation_obstacle_factory.cpp

```diff
@@ -149,7 +149,7 @@ std::vector<ObstaclePtr> RobotNavigationObstacleFactory::createFromTeam(
 ObstaclePtr RobotNavigationObstacleFactory::createFromBallPosition(
     const Point &ball_position) const
 {
-    return createFromShape(Circle(ball_position, BALL_MAX_RADIUS_METERS));
+    return createFromShape(Circle(ball_position, BALL_MAX_RADIUS_METERS * 4));
```

Comment:
why was this changed?

Reply:
We normally would perfectly brush up against any object we're avoiding, since we only account for the size of the object plus the robot radius. That value is then multiplied by a small buffer factor to get the total avoid distance / size of the obstacle. In the case of the ball, this meant the obstacle was much to small to actually avoid properly, so we were constantly running into the ball when we shouldn't have been.

My solution at the time was to increase the size of the ball obstacle, since we can't increase the "fudge factor" without affecting other obstacles.

Reply:
Assuming you're talking about `RobotObstacleInflationFactor`, it multiplies ROBOT_MAX_RADIUS_METERS, not the radius of the obstacle, so if it's a problem for ball, it's probably a problem for other obstacles, too

Reply:
Do you have any ideas for a "more proper" fix here or is this fine for now?

Reply:
You can increase `RobotObstacleInflationFactor` to achieve the same result, i.e. solve for x: 0.203 = BALL_MAX_RADIUS_METERS + ROBOT_MAX_RADIUS_METERS * x. Then the new `RobotObstacleInflationFactor` is x

Reply:
New inflation factor would be 2.0167 by that calculation. This change causes a lot of other obstacle factory tests to fail (not unexpected). Although generally bigger obstacles seems desirable here based on the symptom and reason for the change, I'm not entirely sure that the inflation factor is the best way to address this problem.

It seems like fundamentally we make the "base" obstacle the exact shape of the physical object (eg. a circle of exactly the ball's radius), and then expand that based on the inflation factor. The scaling factor will inflate larger obstacles more than smaller obstacles, meaning if we make this change to make smaller obstacles "work" we may be over-expanding our larger obstacles. This is part of the reason I'm not a fan of using "scaling factors" here, but this is out of scope for this PR (happy to have offline discussion or open an issue where we can talk about this, just wanted to bring it up here).

Do you have a preference for changing the scaling factor vs the "base obstacle shape"? I'll do whichever is preferred for now since fundamentally changing too much else is out of scope here.

Reply:
changed the scaling factor